presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkDiskBasedBroadcastDependency.java

```diff
@@ -0,0 +1,117 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark;
+
+import com.facebook.airlift.log.Logger;
+import com.facebook.presto.spark.classloader_interface.PrestoSparkStorageHandle;
+import com.facebook.presto.spi.storage.TempDataOperationContext;
+import com.facebook.presto.spi.storage.TempStorage;
+import com.facebook.presto.spi.storage.TempStorageHandle;
+import io.airlift.units.DataSize;
+import org.apache.spark.SparkException;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.broadcast.Broadcast;
+import scala.Tuple2;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.List;
+import java.util.concurrent.TimeoutException;
+
+import static com.facebook.presto.ExceededMemoryLimitException.exceededLocalBroadcastMemoryLimit;
+import static com.facebook.presto.spark.util.PrestoSparkUtils.computeNextTimeout;
+import static io.airlift.units.DataSize.succinctBytes;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.stream.Collectors.toList;
+
+public class PrestoSparkDiskBasedBroadcastDependency
+        implements PrestoSparkBroadcastDependency<PrestoSparkStorageHandle>
+{
+    private static final Logger log = Logger.get(PrestoSparkDiskBasedBroadcastDependency.class);
+
+    private final RddAndMore<PrestoSparkStorageHandle> broadcastDependency;
+    private final DataSize maxBroadcastSize;
+    private final long queryCompletionDeadline;
+    private final TempStorage tempStorage;
+    private final TempDataOperationContext tempDataOperationContext;
+    private Broadcast<List<PrestoSparkStorageHandle>> broadcastVariable;
+
+    public PrestoSparkDiskBasedBroadcastDependency(RddAndMore<PrestoSparkStorageHandle> broadcastDependency, DataSize maxBroadcastSize, long queryCompletionDeadline, TempStorage tempStorage, TempDataOperationContext tempDataOperationContext)
+    {
+        this.broadcastDependency = requireNonNull(broadcastDependency, "broadcastDependency cannot be null");
+        this.maxBroadcastSize = requireNonNull(maxBroadcastSize, "maxBroadcastSize cannot be null");
+        this.tempStorage = requireNonNull(tempStorage, "tempStorage cannot be null");
+        this.tempDataOperationContext = requireNonNull(tempDataOperationContext, "tempDataOperationContext cannot be null");
+        this.queryCompletionDeadline = queryCompletionDeadline;
```

Comment:
nit: let's keep the assignments in order (that's what we are trying to do across the codebase)